pkgs/test/lib/src/runner/browser/chrome.dart

```diff
@@ -149,15 +149,28 @@ class Chrome extends Browser {
 }
 
 Future<WipConnection> _connect(
-    Process process, int port, Map<String, String> idToUrl) async {
+    Process process, int port, Map<String, String> idToUrl, Uri url) async {
   // Wait for Chrome to be in a ready state.
   await process.stderr
       .transform(utf8.decoder)
       .transform(LineSplitter())
       .firstWhere((line) => line.startsWith('DevTools listening'));
 
   var chromeConnection = ChromeConnection('localhost', port);
-  var tab = (await chromeConnection.getTabs()).first;
+  ChromeTab tab;
+  var attempt = 0;
+  while (tab == null) {
```

Comment:
nit: add `attempt < 5` and then do the null check/throw after the while - I think that would be a bit cleaner

Reply:
That actually ends up being slightly more verbose. Gonna leave as is.